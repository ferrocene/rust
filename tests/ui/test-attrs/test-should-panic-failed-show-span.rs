//@ compile-flags: --test
//@ run-flags: --test-threads=1 --nocapture
//@ run-fail
//@ check-run-results
//@ exec-env:RUST_BACKTRACE=0
//@ normalize-stdout: "finished in \d+\.\d+s" -> "finished in $$TIME"
//@ needs-threads
//@ needs-unwind (panic)

#[test]
#[should_panic]
fn should_panic_with_any_message() {
    panic!("Panic!");
}

#[test]
#[should_panic = "message"]
fn should_panic_with_message() {
    panic!("message");
}

#[test]
#[should_panic]
fn should_panic_with_any_message_does_not_panic() {
    // DON'T PANIC
}

#[test]
#[should_panic = "message"]
fn should_panic_with_message_does_not_panic() {
    // DON'T PANIC
}

#[test]
#[should_panic = "message"]
fn should_panic_with_substring_panics_with_incorrect_string() {
    panic!("ZOMGWTFBBQ");
}
