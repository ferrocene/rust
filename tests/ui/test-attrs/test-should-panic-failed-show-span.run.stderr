
thread 'should_panic_with_any_message' panicked at $DIR/test-should-panic-failed-show-span.rs:13:5:
Panic!
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

thread 'should_panic_with_message' panicked at $DIR/test-should-panic-failed-show-span.rs:19:5:
message

thread 'should_panic_with_substring_panics_with_incorrect_string' panicked at $DIR/test-should-panic-failed-show-span.rs:37:5:
ZOMGWTFBBQ
